torch/serialization.py

```diff
@@ -453,14 +453,28 @@ def persistent_load(saved_id):
             raise RuntimeError("Unknown saved id type: %s" % saved_id[0])
 
     f_is_real_file = _is_real_file(f)
-    if f_is_real_file and f.tell() == 0:
-        # legacy_load requires that f has fileno()
-        # only if offset is zero we can attempt the legacy tar file loader
-        try:
-            return legacy_load(f)
-        except tarfile.TarError:
-            # if not a tarfile, reset file offset and proceed
-            f.seek(0)
+
+    def raise_load_err_msg(pattern, e):
+        if pattern in str(e):
+            msg = (str(e) + ". You can only torch.load from a file that is seekable." +
+                            " Please pre-load the data into a buffer like io.BytesIO and try to load from it instead.")
+            raise type(e)(msg)
+        else:
+            raise e
+
+    try:
+        if f_is_real_file and f.tell() == 0:
+            # legacy_load requires that f has fileno()
+            # only if offset is zero we can attempt the legacy tar file loader
+            try:
+                return legacy_load(f)
+            except tarfile.TarError:
+                # if not a tarfile, reset file offset and proceed
+                f.seek(0)
+    except io.UnsupportedOperation as e:
+        raise_load_err_msg("seek", e)
+    except AttributeError as e:
```

Comment:
This still makes me nervous. I'd much rather this AttributeError was around the `tell` call specifically, rather than catching everything.